src/service/angular-sails.js

```diff
@@ -1,74 +1,238 @@
-/*jslint sloppy:true*/
-/*global angular, io */
-angular.module('ngSails').provider('$sails', function () {
-    var provider = this,
-        httpVerbs = ['get', 'post', 'put', 'delete'],
-        eventNames = ['on', 'once'];
+(function(angular, io) {
+  'use strict';
+  io.sails.autoConnect = false;
 
+  // copied from angular
+  function parseHeaders(headers) {
+    var parsed = Object.create(null), key, val, i;
```

Comment:
Why the need for `Object.create`? I would go with a "standard" object with the common prototype, it would probably be expected.  If someone really wants, they can use `hasOwnProperty` to filter out the prototype